orbit-love/orbit-browser-extension
# [Developer Experience] Enable hot-reloading in development

## Description

This PR enables hot-reloading in development, meaning one does not need to click the “Update” button in `chrome://extensions` after a code change to see the changes in action.

It uses the fabulous [Web Extensions](https://parceljs.org/recipes/web-extension/) feature of Parcel 2.

## How to test

- Checkout this branch
- Run `yarn` to install the `@parcel/config-webextension` dependency
- Clean up existing artefacts to make sure the branch works on its own: `rm -rf dist extension .parcel-cache`
- Run `yarn start`, and install the extension locally. **Important: the folder to load is now `dist`, not `extension`.** Make sure that everything works as usual.
- Make a change to `options.html` while on the extension options page. The change should be visible right away, or after a page refresh.
- Make a change to the widget while looking at a pull request. The change should be visible right away, or after a page refresh.
- Run `yarn build` to build the artefact
- Load the built artefact (under `dist`) and make sure that everthing works as usual.
